examples/onepanel_submitter.py

```diff
@@ -0,0 +1,172 @@
+import logging
+import pyaml
+import os
+import json
+import zlib
+from couler.argo_submitter import ArgoSubmitter
+import onepanel.core.api
+from onepanel.core.api.rest import ApiException
+
+
+class OnepanelException(Exception):
+    """Base-class for all exceptions raised by the OnepanelSubmitter"""
+
+
+class MissingTokenException(OnepanelException):
+    """
+    Happens when there is no provided token and one can't be found on the system.
+    """
+
+
+class MissingHostException(OnepanelException):
+    """
+    Happens when there is no provided host and can't be found on the system
+    """
+
+
+class InvalidCredentialsException(OnepanelException):
+    """
+    Happens when the credentials to get the authentication token are incorrect.
+    """
+
+
+class LongWorkflowNameException(OnepanelException):
+    """
+    Happens when the workflow name is too long - which is >= 20 characters.
+    """
+
+
+class OnepanelSubmitter(ArgoSubmitter):
+    """A submitter which submits a workflow to Onepanel"""
+    def __init__(self, workflow_name, username=None, host=None, token=None):
+        """
+        username defaults to 'admin'
+        """
+
+        if len(workflow_name) >= 20:
+            raise LongWorkflowNameException('workflow_name must be 20 characters or less.')
+        # If token and api_url are not passed in, use the values from token file and environment variable
+        # If those don't exist, throw an exception
+        self.namespace = os.getenv('ONEPANEL_RESOURCE_NAMESPACE')
+        self.workflow_name = workflow_name
+
+        if username is not None and token is None:
+            raise InvalidCredentialsException('You must provide a token with a username')
+
+        if username is None:
```

Comment:
We do not need this check and should not default to `admin` if `username` is provided because we already have the raw token and the username is irrelevant.